netlib/tcp.py

```diff
@@ -484,12 +484,14 @@ def finish(self):
         if not isinstance(self.connection, SSL.Connection):
             if not getattr(self.wfile, "closed", False):
                 try:
-                    self.wfile.flush()
-                    self.wfile.close()
+                    if self.wfile:
```

Comment:
We are violating `.disconnect()`s precondition here: https://github.com/mitmproxy/mitmproxy/blob/0dbb7033ee1a6e238752381ce99439ef7d38b208/mitmproxy/protocol/base.py#L148-L151.

I think this happens because [this check](https://github.com/mitmproxy/mitmproxy/blob/0dbb7033ee1a6e238752381ce99439ef7d38b208/mitmproxy/proxy/modes/transparent_proxy.py#L24) fails presumably because `.connection` has been set without initializing the object properly (see [ServerConnection.\__bool__](https://github.com/mitmproxy/mitmproxy/blob/0dbb7033ee1a6e238752381ce99439ef7d38b208/mitmproxy/models/connections.py#L48)).